epaint/src/tessellator.rs

```diff
@@ -236,19 +347,26 @@ pub mod path {
     //   - quadrant 3: right top
     // * angle 4 * TAU / 4 = right
     pub fn add_circle_quadrant(path: &mut Vec<Pos2>, center: Pos2, radius: f32, quadrant: f32) {
-        // TODO: optimize with precalculated vertices for some radii ranges
-
-        let n = (radius * 0.75).round() as i32; // TODO: tweak a bit more
-        let n = n.clamp(2, 32);
-        const RIGHT_ANGLE: f32 = TAU / 4.0;
-        path.reserve(n as usize + 1);
-        for i in 0..=n {
-            let angle = remap(
-                i as f32,
-                0.0..=n as f32,
-                quadrant * RIGHT_ANGLE..=(quadrant + 1.0) * RIGHT_ANGLE,
-            );
-            path.push(center + radius * Vec2::angled(angle));
+        if radius == 0.0 {
```

Comment:
```suggestion
        if radius <= 0.0 {
```

seems like a safe fallback when a user has accidentally given a negative radius